packages/block-editor/src/store/selectors.js

```diff
@@ -1389,6 +1389,90 @@ export function canInsertBlocks( state, clientIds, rootClientId = null ) {
 	);
 }
 
+/**
+ * Determines if the given block is allowed to be deleted.
+ *
+ * @param {Object}  state        Editor state.
+ * @param {string}  clientId     The block client Id.
+ * @param {?string} rootClientId Optional root client ID of block list.
+ *
+ * @return {boolean} Whether the given block is allowed to be removed.
+ */
+export function canRemoveBlock( state, clientId, rootClientId = null ) {
+	const attributes = getBlockAttributes( state, clientId );
+
+	// attributes can be null if the block is already deleted.
+	if ( attributes === null ) {
+		return;
+	}
```

Comment:
It seems that `canRemoveBlocks` and `canMoveBlocks` run after a block is deleted (I'm not sure why) causing `attributes` to be null, this is a defensive check.